### ISSUE TITLE
[Bug] Failure providers issues

### ISSUE THREAD

**Issue by burekas7** [elgatito/plugin.video.elementum#644]
This issue includes several issues regarding failure providers:
---

## 1) Why failure providers prevent from the other results to show them?
Steps to Reproduce:
a. Set in Burst a provider which its url is not responding or something, in addition to some other providers that is working for sure.
b. Run a search for a some episode. For example I search for "Westworld S02E03".
c. Here for example for an unknown reason "Idope" makes problems.
(Even though its search url from the log was checked manually in a browser and it seems that it's working, and also I think also in the log it shows that it did return results)
d. I get the "Playback failed" dialog instead of getting the results windows with all the other provider results that are working well.
e. Also the notification of "1 provider left" is still shown after the dialog is closed, this is until the timeout is finished and then it shows which is the problematic providers.

So to sum up:
1) Why I can't see the other results even some of the providers are not working?
2) Why Idope isn't working even the url is working and return results? (According to the log here)

Kodi debug log:
https://paste.ubuntu.com/p/SPMcTJFYNT

Images:
[d] ![image](https://user-images.githubusercontent.com/13424257/99381960-779eb680-28d4-11eb-8d2c-ce55e7bdf982.png)
[e] ![image](https://user-images.githubusercontent.com/13424257/99381855-5047e980-28d4-11eb-80f2-0742d7c31b58.png)
[e] ![image](https://user-images.githubusercontent.com/13424257/99381904-5dfd6f00-28d4-11eb-92ff-6cbf2de7f914.png)

---

## 2) I have a request regarding this issue:
Is there any way to add kind of a sanity test that will run over all the providers and report back a list of which of them are not responding or they have kind of a problem with them?
It will help to disable them in Burst or report it to you to fix them.

Thanks.

---

## Your Environment
Elementum: 0.1.62
Burst: 0.0.50
Kodi: 17.6
Windows 10, 64bit

**Comment by elgatito** [comment]
> Why I can't see the other results even some of the providers are not working?
> Just ignore them and show the rest ot the results and in addition show a dialog/notification which of the providers didn't work/respond.

It is already working in that way. If some providers do not return returns before timeout - it is skipped.
Unless you set some manual timeouts in Elementum for providers or Burst for providers.

There should be an option to skip failed providers in Burst, it is disabled by default.

In terms of Elementum - whole Burst is 1 provider.
Inside of Burst request is splitted per internal providers, and you see the message counting what is left.

**Comment by elgatito** [comment]
> Is there any way to add kind of a sanity test that will run over all the providers and report back a list of which of them are not responding or they have kind of a problem with them?

Even if there was such a check, it should be addressed to Burst, to check existing providers. 
For checking private providers then Burst should have all the accounts enabled, which is not possible.

**Comment by burekas7** [comment]
> It is already working in that way. If some providers do not return returns before timeout - it is skipped.
> Unless you set some manual timeouts in Elementum for providers or Burst for providers.
> 
> There should be an option to skip failed providers in Burst, it is disabled by default.
> 
> In terms of Elementum - whole Burst is 1 provider.
> Inside of Burst request is splitted per internal providers, and you see the message counting what is left.

1) Just to make it clear:
* In Burst there is "Search Timout" (which should be lower the Elementum), this timeout is for all the providers together to finish or it this is a timeout per provider?
* And for Elementum "Providers->Timeout" does it mean for all the providers together to finish?

As I see, in my case "Search Timeout (Burst)" was on 90,
While "Providers->Timeout  (Elementum" was on 60.
So probably this was the reason that the results window didn't popup.
I switched the values and not it's ok and even there is still a problem with idope the results windows popup.

* But I still has a problem with idope even though its url is working, I will open a new issue in Burst.

2) Question:
In Elementum there is also "Disable providers failing health check", it was enabled with 3 tries.
What does it actually do when it is enabled? 
Does it scan all my selected providers for each searching before it start and disable them?
It seems that it didn't work in the log I posted. Or there was another issue with idope as I said, the search url worked so I don't know whey I get a problem with it. Maybe timeout?

**Comment by burekas7** [comment]
> Even if there was such a check, it should be addressed to Burst, to check existing providers.
> For checking private providers then Burst should have all the accounts enabled, which is not possible.

No, I didn't mean to check all the providers in Burst, just those you selected and enabled them.
So even if you have a private provider it has its certification values.

**Comment by burekas7** [comment]
btw, in the debug log you mention the timer like this:
`DEBUG: [script.elementum.burst] Timer: 59s / 60s`
Without mention which provider the timer is related to, and there are several timeouts which run in parallel for each provider.
Just after all the timers I saw this:
`WARNING: [script.elementum.burst] idope timed out`

So I think it would be nice to write the log something like this:
`DEBUG: [script.elementum.burst] [idope] Timer: 59s / 60s`

* I know there is also this line for each provider:
NOTICE: [script.elementum.burst] [thepiratebay] >>      The Pirate Bay returned 45 results in 19.1 seconds, sending 25 best ones

**Comment by elgatito** [comment]
> In Burst there is "Search Timout" (which should be lower the Elementum), this timeout is for all the providers together to finish or it this is a timeout per provider?
> And for Elementum "Providers->Timeout" does it mean for all the providers together to finish?

I have said before. Burst for Elementum is a provider. Trackers inside Burst are providers for Burst, not Elementum. Disable bad provider is about disabling Burst if it does not result results before timeout.
There is a checkbox to be sure timeout in Burst is not bigger than in Elementum. By default it is enabled.

**Comment by antonsoroko** [comment]
@burekas7 this does not look like issue/request for elementum. although idea is ok (if i got it correctly).

so, if you want to propose to implement "disable provider if it fails several time" technique (like in elementum for "final results" providers) in burst for trackers - i think it should be placed to burst repo.
but then you might find yourself in situation when tracker/provider will be disabled b/c of temporary network issue and then you will have to manually enable again. if network issue is local - then all your enabled trackers will be disabled.
and if we will automatically re-enable them let's say every start of kodi, then really dead tracker will always give you issues first N searched.
so, you just need to come up with a good design for this feature request.
